postgres/datadog_checks/postgres/metadata.py

```diff
@@ -106,12 +233,221 @@ def report_postgres_metadata(self):
+            rows = cursor.fetchall()
+            table_info = [dict(row) for row in rows]
+            return self._sort_and_limit_table_info(cursor, dbname, table_info, limit)
+
+        else:
+            # Config error should catch the case where schema collection is enabled
+            # and relation metrics aren't, but adding a warning here just in case
+            self._check.log.warning("Relation metrics are not configured for {dbname}, so tables cannot be collected")
+
+    def _sort_and_limit_table_info(
+        self, cursor, dbname, table_info: List[Dict[str, Union[str, bool]]], limit: int
+    ) -> List[Dict[str, Union[str, bool]]]:
+        def sort_tables(info):
+            cache = self._check.metrics_cache.table_activity_metrics
+            # partition master tables won't get any metrics reported on them,
+            # so we have to grab the total partition activity
+            # note: partitions don't exist in V9, so we have to check this first
+            if VersionUtils.transform_version(str(self._check._version))['version.major'] == "9":
+                return (
+                    cache[dbname][info['name']]['postgresql.index_scans']
+                    + cache[dbname][info['name']]['postgresql.seq_scans']
+                )
+            if not info["has_partitions"]:
+                return (
+                    cache[dbname][info['name']]['postgresql.index_scans']
+                    + cache[dbname][info['name']]['postgresql.seq_scans']
+                )
+            else:
+                # get activity
+                cursor.execute(PARTITION_ACTIVITY_QUERY.format(parent=info['name']))
+                row = cursor.fetchone()
+                return row['total_activity']
+
+        # if relation metrics are enabled, sorted based on last activity information
+        table_info = sorted(table_info, key=sort_tables, reverse=True)
+        return table_info[:limit]
+
+    def _query_table_information_for_schema(
+        self, cursor: psycopg.cursor, schemaname: str, dbname: str
+    ) -> List[Dict[str, Union[str, Dict]]]:
+        """
+        Collect table information per schema. Returns a list of dictionaries
+        with key/values:
+            "id": str
+            "name": str
+            "owner": str
+            "foreign_keys": dict (if has foreign keys)
+                name: str
+                definition: str
+            "indexes": dict (if has indexes)
+                name: str
+                definition: str
+            "columns": dict
+                name: str
+                data_type: str
+                default: str
+                nullable: bool
+            "toast_table": str (if associated toast table exists)
+            "partition_key": str (if has partitions)
+            "num_partitions": int (if has partitions)
+        """
+        self._config.schemas_metadata_config.get('max_tables', 1000)
```

Comment:
I think there's something missing here. You probably meant to save this in a variable and use it on the next line?